cu29_log_runtime/src/lib.rs

```diff
@@ -1,207 +1,248 @@
-use cu29_clock::{ClockProvider, RobotClock};
-use cu29_intern_strs::read_interned_strings;
+use std::fmt::{Debug, Formatter};
+use std::fs::File;
+use std::sync::{Mutex, OnceLock};
+use std::io::{BufWriter, Write};
+use cu29_clock::RobotClock;
 use cu29_log::CuLogEntry;
 use cu29_traits::{CuResult, WriteStream};
-use kanal::{bounded, Sender};
+#[cfg(debug_assertions)]
 use log::{Log, Record};
-use once_cell::sync::OnceCell;
 use std::path::PathBuf;
+#[cfg(debug_assertions)]
 use std::sync::Arc;
-use std::thread;
-use std::thread::{sleep, JoinHandle};
-use std::time::Duration;
+use bincode::config::Configuration;
+use bincode::enc::{Encoder, EncoderImpl};
+use bincode::error::EncodeError;
+use bincode::enc::write::Writer;
+use bincode::enc::Encode;
+#[cfg(debug_assertions)]
+use cu29_intern_strs::read_interned_strings;
 
-// The logging system is basically a global queue.
-static QUEUE: OnceCell<Sender<CuLogEntry>> = OnceCell::new();
```

Comment:
Has this change been run in a benchmark with producers from multiple threads? The two examples are single-threaded producers and I'm wondering if the queue overhead makes it worth it in that scenario.

With a locked logger, callers will block on accessing it but if accessing it is quick, then it may make sense to keep it that way. But if many log producers are contending for the lock, it's possible a mpsc queue would be faster

Reply:
```rust
    for i in 0..1_000_000 {
        info!("This is the logline {} associated with the log Logging and some more = {}, {}", i , i+2, i+3);
    }
```

this does test raw throughput but I'm thinking that it'd be useful to model logging frequency that's not as extreme.

Maybe log every 250us/1ms for a 10ms period, sleep for a bit, then do it again. And create N producers with this configurable behavior. That's certainly more implementation behavior, but I think it'd help with predicting future performance

Reply:
I threw together some code to show what i mean

[0001-Multi-threaded-producer.patch](https://github.com/user-attachments/files/16574343/0001-Multi-threaded-producer.patch)

Thoughts?

Reply:
I agree we need to test the contention too. Can you make just another pair of executable to test them vs normal text logging?

Reply:
Yeah I can make it into a separate benchmark in a follow up PR